### PR TITLE
feat: implement creating and activating wallet with henesis api service

### PR DESCRIPTION
**Pull request by hihiboss**

### CODE REVIEW

File: packages/enclave/src/controllers/btc/wallets.controller.ts [hunk 1]
```diff
@@ -227,4 +242,21 @@ export default class WalletsController
...
   }
+
+  private async createInactiveMasterWallet(
+    req: express.Request
+  ): Promise<InactiveMasterWallet> {
+    const type: string = <string>req.query.type;
+    if (type == undefined || type.toUpperCase() != WalletStatus.INACTIVE) {
+      throw new HenesisError(`type should be inactive`, HttpStatus.BAD_REQUEST);
+    }
+    return await req.sdk.btc.wallets.createInactiveMasterWallet(req.body.name);
```
Comment (JhChoy) [inline packages/enclave/src/controllers/btc/wallets.controller.ts:253]:
```suggestion
  private createInactiveMasterWallet(
    req: express.Request
  ): Promise<InactiveMasterWallet> {
    const type: string = <string>req.query.type;
    if (type == undefined || type.toUpperCase() != WalletStatus.INACTIVE) {
      throw new HenesisError(`type should be inactive`, HttpStatus.BAD_REQUEST);
    }
    return req.sdk.btc.wallets.createInactiveMasterWallet(req.body.name);
```

File: packages/enclave/src/controllers/eth/wallets.controller.ts [hunk 2]
```diff
@@ -223,7 +236,19 @@ export default class WalletsController
...
+  ): Promise<EthMasterWalletData | InactiveMasterWallet> {
+    const type: string = <string>req.query.type;
+    if (type != undefined) {
+      if (type.toUpperCase() != WalletStatus.INACTIVE) {
+        throw new HenesisError(
+          `type '${type}' is not supported`,
+          HttpStatus.BAD_REQUEST
+        );
+      }
+      return await req.sdk.eth.wallets.createInactiveMasterWallet(
```
Comment (JhChoy) [inline packages/enclave/src/controllers/eth/wallets.controller.ts:248]:
```suggestion
      return req.sdk.eth.wallets.createInactiveMasterWallet(
```

File: packages/core/src/btc/wallets.ts [hunk 3]
```diff
@@ -153,4 +154,31 @@ export class BtcWallets extends Wallets<BtcMasterWallet> {
       this.env
     );
   }
+
+  async createInactiveMasterWallet(
+    name: string
+  ): Promise<InactiveMasterWallet> {
+    checkNullAndUndefinedParameter({ name });
+    const encryptionKeyBuffer: Buffer = this.createEncryptionKey("dummy");
```
Comment (AgwaB) [inline packages/core/src/btc/wallets.ts:162]: how about make readonly DUMMY variable to use "dummy" ?

**Review by JhChoy — APPROVED** [review]

File: packages/core/src/wallet.ts [hunk 4]
```diff
@@ -118,6 +118,62 @@ export enum WalletStatus {
...
+    this.name = name;
+    this.blockchain = blockchain;
+    this.henesisKey = henesisKey;
+    this.status = status;
+    this.createdAt = createdAt;
+    this.updatedAt = updatedAt;
+  }
+}
+
+export class ActivatingMasterWallet {
```
Comment (AgwaB) [inline packages/core/src/wallet.ts:149]: how about making it unity with `InactiveMasterWallet`? i sugget ActiveMasterWallet

File: packages/enclave/src/controllers/eth/wallets.controller.ts [hunk 5]
```diff
@@ -223,7 +236,17 @@ export default class WalletsController
 
   private async createMasterWallet(
     req: express.Request
-  ): Promise<EthMasterWalletData> {
+  ): Promise<EthMasterWalletData | InactiveMasterWallet> {
+    const type: string = <string>req.query.type;
+    if (type != undefined) {
+      if (type.toUpperCase() != WalletStatus.INACTIVE) {
```
Comment (AgwaB) [inline packages/enclave/src/controllers/eth/wallets.controller.ts:242]:
```suggestion
    if (type != undefined && type.toUpperCase() != WalletStatus.INACTIVE) {
```

File: packages/enclave/src/controllers/klay/wallets.controller.ts [hunk 6]
```diff
@@ -213,7 +226,17 @@ export default class WalletsController
 
   private async createMasterWallet(
     req: express.Request
-  ): Promise<EthMasterWalletData> {
+  ): Promise<EthMasterWalletData | InactiveMasterWallet> {
+    const type: string = <string>req.query.type;
+    if (type != undefined) {
+      if (type.toUpperCase() != WalletStatus.INACTIVE) {
```
Comment (AgwaB) [inline packages/enclave/src/controllers/klay/wallets.controller.ts:232]:
```suggestion
    if (type != undefined && type.toUpperCase() != WalletStatus.INACTIVE) {
```

File: packages/core/src/btc/wallets.ts (hunk 3 above)
Comment (hihiboss) [reply packages/core/src/btc/wallets.ts]: I've changed this to `createDummyEncryptionKey()` method. It makes `ffffff...` having length 64.

File: packages/core/src/wallet.ts (hunk 4 above)
Comment (hihiboss) [reply packages/core/src/wallet.ts:151]: When activating master wallet, its status is changed to `CREATING`, not `ACTIVE`. I've added some comments above this.

**Review by AgwaB — APPROVED** [review]

**Pull request merged**